src/coreclr/jit/importer.cpp

```diff
@@ -7026,46 +7025,37 @@ bool Compiler::impCanPInvokeInlineCallSite(BasicBlock* block)
 void Compiler::impCheckForPInvokeCall(
     GenTreeCall* call, CORINFO_METHOD_HANDLE methHnd, CORINFO_SIG_INFO* sig, unsigned mflags, BasicBlock* block)
 {
-    CorInfoUnmanagedCallConv unmanagedCallConv;
+    CorInfoCallConvExtension unmanagedCallConv;
 
     // If VM flagged it as Pinvoke, flag the call node accordingly
     if ((mflags & CORINFO_FLG_PINVOKE) != 0)
     {
         call->gtCallMoreFlags |= GTF_CALL_M_PINVOKE;
     }
 
-    if ((sig->flags & CORINFO_SIGFLAG_SUPPRESS_GC_TRANSITION) != 0)
-    {
-        call->gtCallMoreFlags |= GTF_CALL_M_SUPPRESS_GC_TRANSITION;
-    }
-
     if (methHnd)
     {
         if ((mflags & CORINFO_FLG_PINVOKE) == 0)
         {
             return;
         }
 
-        unmanagedCallConv = info.compCompHnd->getUnmanagedCallConv(methHnd);
+        bool suppressGCTransition = false;
+        unmanagedCallConv = info.compCompHnd->getUnmanagedCallConv(methHnd, nullptr, &suppressGCTransition);
+        if (suppressGCTransition)
+        {
+            call->gtCallMoreFlags |= GTF_CALL_M_SUPPRESS_GC_TRANSITION;
+        }
```

Comment:
This should be after the `if { } else { }` so that it applies to both calls and calli.